### PR TITLE
Update bip45 based on comments by dskloet

### PR DESCRIPTION
**Pull request by maraoz**
- fix examples
- fix address discovery section
- fix purpose to 45
- mention difference with bip-0044

### CODE REVIEW

File: bip-0045.mediawiki [hunk 1]
```diff
@@ -165,27 +165,22 @@ protocol to accept or reject it, belong to another BIP, in my opinion.
...
-start to discover the accounts in the following manner:
+start to discover the addresses in the following manner:
 
-# derive the first account's node (index = 0)
-# derive the external chain node of this account
-# scan addresses of the external chain; respect the gap limit described below
-# if no transactions are found on the external chain stop discovery
-# if there are some transactions, increase the account index and go to step 1
+# for each cosigner:
+# derive the cosigner's node (`m / purpose' / cosigner_index`)
```
Comment (dskloet) [inline bip-0045.mediawiki]: Above you replaced $purpose with 45'. Do you want to do that here as well?

File: bip-0045.mediawiki [hunk 2]
```diff
@@ -165,27 +165,22 @@ protocol to accept or reject it, belong to another BIP, in my opinion.
...
+# for both the external and internal chains on this node (`m / purpose' / cosigner_index / 0` and `m / purpose' / cosigner_index / 1`)
```
Comment (dskloet) [inline bip-0045.mediawiki]: end with a colon as on line 170:

File: bip-0045.mediawiki [hunk 3]
```diff
@@ -165,27 +165,22 @@ protocol to accept or reject it, belong to another BIP, in my opinion.
...
+# scan addresses of the chain; respect the gap limit described below
 
-This algorithm is correct, because software should disallow creation of new
-accounts if previous one has no transaction history as described in chapter
-"Account" above.
 
-Please note that the algorithm works with the transaction history, not account
-balances, so you can have account with total 0 coins and the algorithm will
-still continue with discovery.
+Please note that the algorithm uses the transaction history, not address
```
Comment (dskloet) [inline bip-0045.mediawiki]: Did you mean to keep 2 blank lines above this one?

File: bip-0045.mediawiki [hunk 4]
```diff
@@ -165,27 +165,22 @@ protocol to accept or reject it, belong to another BIP, in my opinion.
...
-# derive the external chain node of this account
-# scan addresses of the external chain; respect the gap limit described below
-# if no transactions are found on the external chain stop discovery
-# if there are some transactions, increase the account index and go to step 1
+# for each cosigner:
+# derive the cosigner's node (`m / purpose' / cosigner_index`)
+# for both the external and internal chains on this node (`m / purpose' / cosigner_index / 0` and `m / purpose' / cosigner_index / 1`)
+# scan addresses of the chain; respect the gap limit described below
 
-This algorithm is correct, because software should disallow creation of new
```
Comment (dskloet) [inline bip-0045.mediawiki]: If you want, you can put in a reminder that, opposite to BIP44, each cosigner branch needs to be checked, even if the earlier ones don't have transactions.

File: bip-0045.mediawiki (hunk 1 above)
Comment (maraoz) [inline bip-0045.mediawiki]: yeah, will fix

**Comment by laanwj** [comment]
Let me know when this is ready for merging

**Comment by maraoz** [comment]
Addressed all comments by @dskloet. When he ACKs, we can merge. Thanks @laanwj :+1:

**Comment by dskloet** [comment]
LGTM

**Pull request merged**